monai/transforms/utility/dictionary.py

```diff
@@ -1740,6 +1742,57 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> dict[Hashable, N
         return d
 
 
+class CoordinateTransformd(MapTransform, InvertibleTransform):
```

Comment:
Hi @KumoLiu,
As we discussed today in regards to https://github.com/Project-MONAI/tutorials/pull/1795 I'd propose changing the name of this class and it's operations to make it clearer what's going on. What the class does is take the affine from whatever `refer_key` references and applying that to those in `keys` which for our purposes at the moment are points in space. I would recommend a name like `ApplyTransformToPointsd` or just `ApplyTransformd` if we want to add other things besides points in the future (normals, for example). 

I would also not have a `mode` keyword but instead a boolean stating whether the affine is applied or the inverse of the affine. If `inverse` is True instead of its default False, the inverse of the affine is applied to points (and so of course the `inverse()` operation applies the original affine). I think it helps to be explicit about what the transform is doing rather tha using "image-to-world" and other terms, this was what confused me a bit earlier. If people have reason to apply transforms in some bizarre order this makes it clear what is being done so they can choose what is a forward or inverse transformation. 

I'm not sure if we can do something about the `affine_lps_to_ras` argument here which doesn't seem clean here. I know what when loading Nifti files we fix the header and can then use `Orientationd` to fix the orientation, can we not do something with ITK-loaded images and not need this value? You've looked into this longer than I have so I may be off here, if it's harder to fix we can come back to this later.

It might also be helpful if in this class and the array version (whatever we end calling it) we can explicitly specify the affine to apply, either as a provided `__init__` or `__call__` argument, or as a dictionary value referred to by key. 

Thanks!

Reply:
Hi @ericspod, Thank you for all the suggestions. I try to address all of them.

- I renamed the transform to `ApplyTransformToPointsd`, as it currently only applies to points. The name `ApplyTransformd` felt too broad—what do you think?

- I removed the `mode` argument and opted to use `invert_affine` instead. I avoided using `inverse` to prevent conflicts with the `inverse` function.

- I also added `affine` in both the `__call__` and `__init__` methods.

- The `affine_lps_to_ras` parameter is used to ensure that points and images are aligned within the same coordinate system. This parameter is set to True only if the data is loaded by ITK with affine_lps_to_ras=True. For example, if users save annotations (points) using 3DSlicer, they are saved in LPS format. However, when loading images with ITKReader and using affine_lps_to_ras=True, the images are in RAS format. Therefore, we need this flag to ensure both points and images are in the same coordinate system. 
To be honest, it might be better to add this argument in something like LoadPoint, similar to what we do when loading images. However, since we currently don’t have a transform specifically for loading points, could we leave this argument here for now? Once we develop a loading function for geometric data, we can move this argument there.
https://github.com/Project-MONAI/MONAI/blob/a5fbe716378948630783deef8ee435e7e3bdc918/monai/data/image_reader.py#L176

Could you please help review the PR again? I also included the unittests. Thanks again!

Reply:
Let's look into adding `affine_lps_to_ras` to `LoadPoints` or whatever loading class we define, you're right that leaving it here for now makes sense.

Reply:
100 percent agree with this statement "To be honest, it might be better to add this argument in something like `LoadPoint`, similar to what we do when loading images." 
I would say lets leave it here for now. But we should consider expanding the `LoadPoint` definition to like `LoadGeometry` (or something like that), where we can pass an additional parameter like `primitive_type` with options like `box`,  `circle`. I don't think we need to cover anything more. `box`, `circle` and `points` would be sufficient.